### PR TITLE
Мишин И.А.  381803-2.ЛР3

### PR DESCRIPTION
**Pull request by Iliamish**
### Pull Request Readiness Checklist

See details at [code review]( https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/main/lab-guide/lab-1-setup.md#%D1%80%D0%B5%D1%86%D0%B5%D0%BD%D0%B7%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

- [ ] Code is well documented and more information can be seen on the [website](https://devtools-course-practice-2021.readthedocs.io/en/latest/modules.html) (recommended)
- [ ] Make sure that the testing has passed successfully on [GitHub Actions](https://github.com/UNN-ITMM-Software/devtools-course-practice/actions)
- [ ] Get two approve from your fellow students
- [ ] Take a review from one of the administrators

### CODE REVIEW

**Comment by Iliamish** [comment]
@MADZEROPIE @BrandonRTL @andrewkin77 please review my request

**Review by MADZEROPIE — APPROVED** [review]
_**My job here is done**_

File: modules/Ceasar_cipher/src/ceasar_application.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        "Example: " + filename + " -d \"cipher\" -k 2.\n\n";
+}
+
+int checkKey(const char* argv) {
+    int i = 0;
+    for (; argv[i] != '\0'; ++i) {
+        if (argv[i] < '0' || argv[i] > '9') {
+            throw std::runtime_error("ERROR: Wrong number format!\n");
+        }
+    }
```
Comment (MADZEROPIE) [inline modules/Ceasar_cipher/src/ceasar_application.cpp:27]: По моей оригинальной идее, ключ мог быть отрицательным. Но и так сойдёт.

File: modules/Ceasar_cipher/src/ceasar_application.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+#include "include/cipher.h"
+#include "include/ceasar_application.h"
+
+std::string CeasarApplication::getHelp(const std::string& filename) const {
+    return "This is an application that designed for encryption and decryption Caesar chipher " \
+        "for given arguments. \n\n" \
+        "Please provide arguments in the following format:\n$ " +
+        filename + " <-e \"encoded_string\"> or <-d \"decoded_string\"> [-k key {default = 1}] \n\n" \
+        "Example: " + filename + " -d \"cipher\" -k 2.\n\n";
```
Comment (BrandonRTL) [inline modules/Ceasar_cipher/src/ceasar_application.cpp:18]: Надо бы еще пример запуска на encode

**Review by BrandonRTL — APPROVED** [review]

**Review by dmitrybezro — APPROVED** [review]
It`s good :)

**Review by WerWebWer — APPROVED** [review]

**Pull request merged**